### PR TITLE
Improve Rust build/publish

### PR DESCRIPTION
**Pull request by justsmth**
### Issues:
Resolves #ISSUE-NUMBER1
Addresses #ISSUE-NUMBER2

### Description of changes: 
* When Go or Perl are not available, the `aws-lc-sys` crate build fails w/o any comprehensible explanation about the cause of the failure. This change makes the failure cause explicit.
* Add a `publish.sh` script to automate the publishing process. Moved the format/clippy operations into this script. (Script requires additional `-p` argument to actually publish.)
* Place marker file when generation completes to help avoid accidentally publishing after a failed generation attempt.
* Update `aws-lc-sys` crate version to "0.1.1".

### Call-outs:
N/A

### Testing:
I uninstalled `go` from my Mac to verify that the error message displays as expected:
```
➜ cargo test       
   Compiling foreign-types-shared v0.1.1
   Compiling once_cell v1.16.0
   Compiling openssl-sys v0.9.77
   Compiling aws-lc-sys v0.1.1 (/Users/justsmth/repos/aws-lc/bindings/rust/tmp/aws-lc-sys)
   Compiling foreign-types v0.3.2
   Compiling openssl v0.10.42

error: failed to run custom build command for `aws-lc-sys v0.1.1 (/Users/justsmth/repos/aws-lc/bindings/rust/tmp/aws-lc-sys)`

Caused by:
  process didn't exit successfully: `/Users/justsmth/repos/aws-lc/bindings/rust/tmp/aws-lc-sys/target/debug/build/aws-lc-sys-05e8d9a01907a1a7/build-script-build` (exit status: 101)
  --- stderr
  Missing dependency: go-lang
  thread 'main' panicked at 'Required build dependency is missing. Halting build.', build.rs:274:9
  note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
warning: build failed, waiting for other jobs to finish...
```

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license and 
the ISC license.

### CODE REVIEW

File: bindings/rust/generate/generate.sh [hunk 1]
```diff
@@ -147,6 +148,8 @@ function create_prefix_headers {
 
 function prepare_crate_dir {
   echo Preparing crate directory: "${CRATE_DIR}"
+  rm -rf "${CRATE_DIR}"
```
Comment (skmcgrail) [inline bindings/rust/generate/generate.sh:151]: nit: Probably worth leaving a comment that this will also remove the competition marker (which we want).

**Review by skmcgrail — APPROVED** [review]

File: bindings/rust/publish/publish.sh [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+cargo clean
+cargo test # sanity check
+cargo clean
+cargo clippy --fix --allow-no-vcs
+cargo fmt
+cargo publish --dry-run --allow-dirty --no-verify
+
+if [[ ${PUBLISH} -eq 1 ]]; then
+  # The --no-verify is needed because we create `src/bindings.rs` during the build process.
+  cargo publish --allow-dirty --no-verify
```
Comment (samuel40791765) [inline bindings/rust/publish/publish.sh:43]: nit: maybe leaving a comment that specifies the size limit on crate uploads would be helpful for future reference

**Review by skmcgrail — DISMISSED** [review]

File: bindings/rust/generate/generate.sh (hunk 1 above)
Comment (justsmth) [reply bindings/rust/generate/generate.sh:152]: Fixed in latest commit.

**Review by samuel40791765 — DISMISSED** [review]

**Review by samuel40791765 — APPROVED** [review]

File: bindings/rust/publish/publish.sh (hunk 2 above)
Comment (justsmth) [reply bindings/rust/publish/publish.sh:44]: Fixed in latest commit.

**Pull request merged**